### PR TITLE
Issue #12859 Keyboard modifiers support for simulateKeyEvent

### PR DESCRIPTION
**Pull request by haslam22**
Added support for sending additional arguments to simulateKeyEvent, e.g. modifiers such as `ctrlKey: true` or any other value supported by `KeyboardEventInit`.

### CODE REVIEW

**Comment by petetnt** [comment]
Sorry about being so slow with this, been swamped with work as of late. I'll review this on Sunday / Monday. 

Again, apologies for taking so long!

**Comment by haslam22** [comment]
No worries! Thanks for putting aside time to review this.

File: test/spec/SpecRunnerUtils.js [hunk 1]
```diff
@@ -995,20 +995,34 @@ define(function (require, exports, module) {
...
      * @param {HTMLElement} element Element to receive event
+     * @param {KeyboardEventInit} options Optional arguments for key event
      */
-    function simulateKeyEvent(key, event, element) {
-        var doc = element.ownerDocument,
-            oEvent = doc.createEvent('KeyboardEvent');
+    function simulateKeyEvent(key, event, element, options) {
+        var doc = element.ownerDocument;
+
+        if(typeof options === 'undefined') {
```
Comment (petetnt) [inline test/spec/SpecRunnerUtils.js]: We can use `Object.assign` to make this a bit easier:

``` js
var defaultOptions = {
  options.view = doc.defaultView;
  options.bubbles = true;
  options.cancelable = true;
  options.keyIdentifier = key;
};
var newOptions = Object.assign({}, options, defaultOptions});
var oEvent = new KeyboardEvent(event, newOptions);
```

File: test/spec/SpecRunnerUtils.js [hunk 2]
```diff
@@ -995,20 +995,34 @@ define(function (require, exports, module) {
...
+            options.keyIdentifier = key;
+        }
+
+        var oEvent = new KeyboardEvent(event, options);
 
         if (event !== "keydown" && event !== "keyup" && event !== "keypress") {
             console.log("SpecRunnerUtils.simulateKeyEvent() - unsupported keyevent: " + event);
             return;
-        }
+        }
```
Comment (petetnt) [inline test/spec/SpecRunnerUtils.js]: Nit: trailing whitespace

**Review by petetnt — CHANGES_REQUESTED** [review]
Some review comments, mostly looks good to me! Needs an unit test though 👍

**Comment by haslam22** [comment]
Made the requested changes, and provided a unit test. Had to pop it in a new file, `SpecRunnerUtils-test`, hope that's fine. Let me know if anything else needs to be changed, thanks!

File: test/spec/SpecRunnerUtils.js [hunk 3]
```diff
@@ -995,15 +995,21 @@ define(function (require, exports, module) {
...
-            oEvent = doc.createEvent('KeyboardEvent');
+    function simulateKeyEvent(key, event, element, options) {
+        var doc = element.ownerDocument;
+        var defaultOptions = {
+        		view: doc.defaultView,
+        		bubbles: true,
+        		cancelable: true,
+        		keyIdentifier: key
+        };
+        var newOptions = Object.assign({}, options, defaultOptions);
```
Comment (petetnt) [inline test/spec/SpecRunnerUtils.js]: It was brought up in another PR that using `Object.assign` actually brokes the Linux version due to using an age old CEF which doesn't support it 😢 

If you could change this to the previous format then this one is ready to be merged

**Review by petetnt — CHANGES_REQUESTED** [review]

File: test/spec/SpecRunnerUtils.js (hunk 3 above)
Comment (haslam22) [inline test/spec/SpecRunnerUtils.js]: No problem at all, reverted to the old version. 👍

**Comment by petetnt** [comment]
I'll try and test this out for the final time today and then merge it in, LGTM so far :1st_place_medal:

**Comment by haslam22** [comment]
Great, thanks!

File: test/spec/SpecRunnerUtils.js [hunk 4]
```diff
@@ -995,15 +995,28 @@ define(function (require, exports, module) {
      * Simulate key event. Found this code here:
```
Comment (petetnt) [inline test/spec/SpecRunnerUtils.js]: We can remove these comments now as the implementation is different

Comment (petetnt) [inline test/spec/SpecRunnerUtils.js]: We can remove these comments now as the implementation is different

**Review by petetnt — CHANGES_REQUESTED** [review]

**Comment by petetnt** [comment]
Okay, one more nitpick and then it's ready to go 👍  Great work @haslam22!

**Comment by haslam22** [comment]
Woops, forgot to update that part. Should be all good now.

**Review by petetnt — APPROVED** [review]

**Comment by petetnt** [comment]
Thanks for this great contribution @haslam22!

**Pull request merged**